### PR TITLE
HAAR-2229: List all client Ids

### PR DESCRIPTION
**Pull request by ravmoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt [hunk 1]
```diff
@@ -82,6 +86,10 @@ class ClientsService(
       },
     )
   }
+  private val clientSecretGenerator: StringKeyGenerator = Base64StringKeyGenerator(
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt:89]: Has this changed?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt [hunk 2]
```diff
@@ -25,7 +27,9 @@ import uk.gov.justice.digital.hmpps.authorizationserver.utils.OAuthClientSecret
 import java.time.Instant
 import java.time.LocalDate
 import java.time.temporal.ChronoUnit
+import java.util.*
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt:30]: Thought the .* gave lint problem?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt [hunk 3]
```diff
@@ -134,6 +134,12 @@ class ClientsController(
     )
     telemetryClient.trackEvent("AuthorizationServerClientDeploymentDetailsUpsert", telemetryMap)
   }
+
```
Comment (AlTurner-MOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt:137]: I think you added a `MigrationController` under your last ticket - given that this end point should only be required for the migration period, probably better to put it in there? Thinking a `MigrationService` might also be a good idea i.e. to keep all of the temporary code concerned only with migration isolated?

Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientsController.kt]: Thanks, Updated

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**